include/Reader/reader.h

```diff
@@ -3241,6 +3251,13 @@ class ReaderBase {
   virtual bool arrayGet(CORINFO_SIG_INFO *Sig, IRNode **RetVal) = 0;
   virtual bool arraySet(CORINFO_SIG_INFO *Sig) = 0;
 
+  /// \brief Check whether structs are represented by pointers on the operand
+  /// stack.
+  ///
+  /// \returns true iff structs are represented by pointers on the operand
+  /// stack.
+  virtual bool structsAreRepresentedByPointers() = 0;
```

Comment:
I haven't worked this out completely, but if `addressOfValue` is suitably savvy then we perhaps could always call it from `rdrCallWriteBarrierHelper` for the value class case, and (depending on client strategy) it would either just return the address it already has or save the struct to a temp and return that temp's address. Then we don't need to have this info visible at the base layer.

I'd be fine if you leave what you have for this and we can fix this part up later.